exporter/datadogexporter/config.go

```diff
@@ -0,0 +1,111 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package datadogexporter
+
+import (
+	"fmt"
+
+	"go.opentelemetry.io/collector/config/configmodels"
+)
+
+// DogStatsDConfig defines the DogStatsd related configuration
+type DogStatsDConfig struct {
+	// Endpoint is the DogStatsD address.
+	// The default value is 127.0.0.1:8125
+	// A Unix address is supported
+	Endpoint string `mapstructure:"endpoint"`
```

Comment:
Can you use the https://github.com/open-telemetry/opentelemetry-collector/blob/master/config/confignet/confignet.go#L44 or https://github.com/open-telemetry/opentelemetry-collector/blob/master/config/confignet/confignet.go#L22?

Reply:
The `statsd` library picks up which protocol to use automatically (`udp` or `unixgram`) based on the `endpoint` field shape, so the `transport` field would have to be ignored in some cases (in particular, if `endpoint` starts with `unix://` it's considered UDS, and if certain environment variables are set it would be UDP). Would that behavior be okay you? Otherwise we would have to leave it as it is right now.